pkg/utils/functional/functional.go

```diff
@@ -90,16 +90,47 @@ func MergeInto(dest interface{}, srcs ...interface{}) {
 	}
 }
 
-// MergeStringMaps merges all key value pairs into a single map, last write wins.
-func MergeStringMaps(maps ...map[string]string) map[string]string {
+// UnionStringMaps merges all key value pairs into a single map, last write wins.
+func UnionStringMaps(maps ...map[string]string) map[string]string {
 	result := map[string]string{}
 	for _, m := range maps {
 		for k, v := range m {
 			result[k] = v
 		}
 	}
 	return result
+}
+
+// IntersectStringSlice takes the intersection of all string slices
+func IntersectStringSlice(slices ...[]string) []string {
+	// count occurances
+	counts := map[string]int{}
+	for _, strings := range slices {
+		for _, s := range UniqueStrings(strings) {
+			counts[s] = counts[s] + 1
+		}
+	}
+	// select if occured in all
+	var intersection []string
+	for key, count := range counts {
+		if count == len(slices) {
+			intersection = append(intersection, key)
+		}
+	}
+	return intersection
+}
 
+func UniqueStrings(strings []string) []string {
+	exists := map[string]bool{}
+	for _, s := range strings {
+		exists[s] = true
+	}
+	var unique []string
+	for s := range exists {
```

Comment:
Minor optimization, we can do it is a single loop-
```
exists := map[string]bool{}
var unique []string
for _, s := range strings {
        if exists[s] {
              continue
        }
	exists[s] = true
         unique = append(unique, s)
}
return unique
```

Reply:
I'm allergic to continue statements. It unrolls to an identical number of statements at the end of the day.

Reply:
Actually, I think mine might unroll to fewer statements, since there's no if involved.

n * ifstmt
n * exists...
n * unique...

vs 
n * exists...
m * unique...

Not that it matters at all 😄